framework/packages/abstract-client/src/account.rs

```diff
@@ -293,7 +326,46 @@ impl<'a, Chain: CwEnv> AccountBuilder<'a, Chain> {
         verifiers::validate_description(self.description.as_deref())?;
         verifiers::validate_link(self.link.as_deref())?;
 
-        let install_modules = self.install_modules.clone();
+        let funds = self.generate_funds(&install_modules, true)?;
+        let account_details = AccountDetails {
+            name,
+            description: self.description.clone(),
+            link: self.link.clone(),
+            namespace: self.namespace.as_ref().map(ToString::to_string),
+            base_asset: self.base_asset.clone(),
+            install_modules,
+            account_id: self.expected_local_account_id,
+        };
+        let abstract_account = match self.owner_account {
+            None => self.abstr.account_factory.create_new_account(
+                account_details,
+                ownership,
+                Some(&funds),
+            )?,
+            Some(owner_account) => owner_account
+                .abstr_account
+                .create_sub_account(account_details, Some(&funds))?,
+        };
+        Ok(Account::new(abstract_account, self.install_on_sub_account))
+    }
+
+    /// Modules to install without duplicates
+    fn install_modules(&self) -> Vec<ModuleInstallConfig> {
+        let mut install_modules = self.install_modules.clone();
+        if self.ibc_enable.unwrap_or(true) {
+            install_modules.push(IbcClient::<Chain>::install_config(&Empty {}).unwrap());
+        }
+        install_modules.dedup();
```

Comment:
This dedup might not be enough as it will keep two modules of the same kind but different versions. I.e. if I add two DCA apps with different versions, the dedup won't remove one of them.

Reply:
Also it will fail on same module with same version if init message is different. Wasn't aiming to solve all issues of installation, just this basic one